modules/core/src/test/java/org/apache/ignite/internal/processors/metastorage/persistence/DistributedMetaStoragePersistentTest.java

```diff
@@ -84,6 +86,32 @@ public void testRestart() throws Exception {
         assertEquals("value", ignite.context().distributedMetastorage().read("key"));
     }
 
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testDeactivateWriteActivate() throws Exception {
+        startGrid(0);
+
+        grid(0).cluster().state(ClusterState.ACTIVE);
+        grid(0).cluster().state(ClusterState.INACTIVE);
+
+        // Deactivation is asynchronous.
+        Thread.sleep(100);
```

Comment:
Sleep can lead to racing.
Maybe we should wait for event `org.apache.ignite.events.EventType#EVT_CLUSTER_DEACTIVATED`?

Reply:
Good idea